wradlib/georef/raster.py

```diff
@@ -475,3 +479,87 @@ def reproject_raster_dataset(src_ds, **kwargs):
     gdal.ReprojectImage(src_ds, dst_ds, src_srs, dst_srs, resample)
 
     return dst_ds
+
+
+def get_raster_extent(dataset, geo=True, window=True):
+    """Get the coordinates of the 4 corners of the raster dataset
+
+    Parameters
+    ----------
+    dataset : gdal.Dataset
+        raster image with georeferencing (GeoTransform at least)
+    geo : bool
+        True to get geographical coordinates
+    window : bool
+        True to get the extent of the window containing the corners instead
+
+    Returns
+    -------
+    extent : :class:`numpy:numpy.ndarray`
+        corner coordinates [ul,ll,lr,ur] or
+        window extent [xmin, xmax, ymin, ymax]
+    """
+
+    x_size = dataset.RasterXSize
+    y_size = dataset.RasterYSize
+    geotrans = dataset.GetGeoTransform()
+    xmin = geotrans[0]
+    ymax = geotrans[3]
+    xmax = geotrans[0] + geotrans[1] * x_size
```

Comment:
This covers the UPPER LEFT corner but not the LOWER RIGHT corner.

Reply:
I do not understand.

Reply:
If geotrans[0]/geotrans[3] is the UPPER LEFT corner point, the the LOWER RIGHT corner point would be

geotrans[0] + getrans[1] * x_size + x_size
geotrans[3] + getrans[3] * y_size + y_size

Reply:
In a four pixels image with size 1 and UL corner at 0,0 LR corner would be:
0 + 1 * 2 = 2
0 - 1 * 2 = -2
My tests with actual values are also successful.
Am I missing something?

Reply:
@egouden Yes, you are right and I'm wrong. Not sure what I was thinking when I wrote it. Sorry for the confusion...